internal/service/alarm_notification_handler.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2023 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package service
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"net/http"
+	"slices"
+	"sync"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/openshift-kni/oran-o2ims/internal/data"
+	"github.com/openshift-kni/oran-o2ims/internal/jq"
+	"github.com/openshift-kni/oran-o2ims/internal/k8s"
+	"github.com/openshift-kni/oran-o2ims/internal/persiststorage"
+	"github.com/openshift-kni/oran-o2ims/internal/search"
+)
+
+// singlton alarm notification handler
+var singleAlarmNotificationHandle *alarmNotificationHandler = nil
+
+// AlarmNotificationManagerHandlerBuilder contains the data and logic needed to create a new alarm notification
+// collection handler. Don't create instances of this type directly, use the
+// NewAlarmNotificationHandler function instead.
+type alarmNotificationHandlerBuilder struct {
+	logger         *slog.Logger
+	loggingWrapper func(http.RoundTripper) http.RoundTripper
+	cloudID        string
+	extensions     []string
```

Comment:
I believe extensions aren't needed here either. Can you check and remove them if they aren't used?